test/unit/matrix/test_matrix.cpp

```diff
@@ -528,64 +686,112 @@ TYPED_TEST(MatrixTest, Dependencies) {
-      LayoutInfo layout = tileLayout(distribution.localSize(), test.block_size);
-      memory::MemoryView<Type, Device::CPU> mem(layout.minMemSize());
-      const Type* p = mem();
-      Matrix<const Type, Device::CPU> mat(std::move(distribution), layout, p);
-      auto rosenders1 = getReadSendersUsingGlobalIndex(mat);
-      EXPECT_TRUE(checkSendersStep(rosenders1.size(), rosenders1));
+      auto senders0 = getReadWriteSendersUsingLocalIndex(mat);
+      EXPECT_TRUE(checkSendersStep(senders0.size(), senders0));
 
-      auto rosenders2 = getReadSendersUsingLocalIndex(mat);
-      EXPECT_TRUE(checkSendersStep(rosenders2.size(), rosenders2));
+      auto senders1 = getReadWriteSendersUsingGlobalIndex(mat);
+      EXPECT_TRUE(checkSendersStep(0, senders1));
+
+      auto [rosenders2a, rosenders2b, senders3] = [&]() {
+        auto mat_sub = mat.subPipeline();
+
+        auto rosenders2a = getReadSendersUsingLocalIndex(mat_sub);
+        EXPECT_TRUE(checkSendersStep(0, rosenders2a));
+
+        auto rosenders2b = getReadSendersUsingGlobalIndex(mat_sub);
+        EXPECT_TRUE(checkSendersStep(0, rosenders2b));
+
+        auto senders3 = getReadWriteSendersUsingLocalIndex(mat_sub);
+        EXPECT_TRUE(checkSendersStep(0, senders3));
+
+        return std::tuple(std::move(rosenders2a), std::move(rosenders2b), std::move(senders3));
+      }();
+
+      auto rosenders4a = getReadSendersUsingGlobalIndex(mat);
+      EXPECT_TRUE(checkSendersStep(0, rosenders4a));
+
+      CHECK_MATRIX_SENDERS(true, senders1, senders0);
+      EXPECT_TRUE(checkSendersStep(0, rosenders2b));
+      CHECK_MATRIX_SENDERS(true, rosenders2b, senders1);
+      EXPECT_TRUE(checkSendersStep(rosenders2a.size(), rosenders2a));
+
+      CHECK_MATRIX_SENDERS(false, senders3, rosenders2b);
+      CHECK_MATRIX_SENDERS(true, senders3, rosenders2a);
+
+      CHECK_MATRIX_SENDERS(true, rosenders4a, senders3);
+
+      auto rosenders4b = getReadSendersUsingLocalIndex(mat);
+      EXPECT_TRUE(checkSendersStep(rosenders4b.size(), rosenders4b));
+
+      auto senders5 = getReadWriteSendersUsingGlobalIndex(mat);
+      EXPECT_TRUE(checkSendersStep(0, senders5));
+
+      CHECK_MATRIX_SENDERS(false, senders5, rosenders4a);
+      CHECK_MATRIX_SENDERS(true, senders5, rosenders4b);
     }
   }
 }
 
-TYPED_TEST(MatrixTest, DependenciesReferenceMix) {
+TYPED_TEST(MatrixTest, DependenciesSubSubPipeline) {
   using Type = TypeParam;
 
   for (const auto& comm_grid : this->commGrids()) {
     for (const auto& test : sizes_tests) {
       // Dependencies graph:
-      // rw0 - rw1 - ro2a - rw3 - ro4a - rw5
-      //           \ ro2b /    \ ro4b /
+      // rw0 - rw1 - ro2a ------- rw3 - ro4a ------- rw5
+      //           \ ------ ro2b /     \ ----- ro4b /
+      //
+      //             +---------------------+
+      //                  sub pipeline
+      //
+      //                    +-------+
+      //                sub-sub pipeline
```

Comment:
Wait... does this mean that ro2a, ro2b can run concurrently and ro4a, ro4b as well?

Reply:
No, not without https://github.com/eth-cscs/DLA-Future/issues/923... This depicts the theoretical best case. Would you like me to add a comment explicitly stating this or even update the diagram?